packages/language-server/src/server.ts

```diff
@@ -0,0 +1,156 @@
+import {
+    IConnection,
+    TextDocuments,
+    DiagnosticSeverity,
+    Diagnostic,
+    createConnection,
+    IPCMessageReader,
+    IPCMessageWriter
+} from 'vscode-languageserver'
+import { TextDocument } from 'vscode-languageserver-textdocument'
+import * as MessageHandler from './MessageHandler'
+import { fullDocumentRange } from './provider'
+import * as util from './util'
+import lint from './lint'
+import fs from 'fs'
+import install from './install'
+
+export interface LSOptions {
+    /**
+     * If you have a connection already that the ls should use, pass it in.
+     * Else the connection will be created from `process`.
+     */
+    connection?: IConnection;
+}
+
+/**
+ * Starts the language server.
+ *
+ * @param options Options to customize behavior
+ */
+export function startServer(options?: LSOptions) {
+    let connection = options?.connection;
+    if (!connection) {
+        connection = process.argv.includes('--stdio')
+            ? createConnection(process.stdin, process.stdout)
+            : createConnection(new IPCMessageReader(process), new IPCMessageWriter(process));
+    }
+
+    const documents: TextDocuments<TextDocument> = new TextDocuments(TextDocument)
+
+    connection.onInitialize(async () => {
+        const binPathPrismaFmt = await util.getBinPath()
+        if (!fs.existsSync(binPathPrismaFmt)) {
+            try {
+                await install(binPathPrismaFmt)
+                connection!.console.info(
+                    'Prisma plugin prisma-fmt installation succeeded.',
+                )
+            } catch (err) {
+                connection!.console.error('Cannot install prisma-fmt: ' + err)
+            }
+        }
+
+        connection!.console.info(
+            'Installed version of Prisma binary `prisma-fmt`: ' +
+            (await util.getVersion()),
+        )
+
+        const pj = util.tryRequire('../../package.json')
+        connection!.console.info(
+            'Extension name ' + pj.name + ' with version ' + pj.version,
```

Comment:
As `util.tryRequire` might not return anything, this is doomed to fail.
Cannot read property `name` of undefined.

You should either try catch the require here and handle it or deal with the case, that pj might not be there.

Using `any` for the `tryRequire` is the reason the compiler doesn't complain, but that won't guarantee the program to run.